From d78fd9a091f4c73a0bf6b75edf56caa3eabbe63b Mon Sep 17 00:00:00 2001
From: Vadim Rutkovsky <roignac@gmail.com>
Date: Thu, 27 Apr 2017 17:01:48 +0200
Subject: [PATCH] Conductor image: support fedora (#460)

* Conductor image: support fedora

* Include 64-bits libs path in LD_LIBRARY_PATH

This fixes missing library errors on fedora images
---
 container/core.py                                  | 2 +-
 container/docker/templates/conductor-dockerfile.j2 | 8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/container/core.py b/container/core.py
index ca2b2af..4535dca 100644
--- a/container/core.py
+++ b/container/core.py
@@ -670,7 +670,7 @@ def conductorcmd_build(engine_name, project_name, services, cache=True,
                 if not local_python:
                     # Use the conductor's Python runtime
                     run_kwargs['environment'] = dict(
-                         LD_LIBRARY_PATH='/_usr/lib:/_usr/local/lib',
+                         LD_LIBRARY_PATH='/_usr/lib:/_usr/lib64:/_usr/local/lib',
                          CPATH='/_usr/include:/_usr/local/include',
                          PATH='/usr/local/sbin:/usr/local/bin:'
                               '/usr/sbin:/usr/bin:/sbin:/bin:'
diff --git a/container/docker/templates/conductor-dockerfile.j2 b/container/docker/templates/conductor-dockerfile.j2
index 734c2cb..7c53c92 100644
--- a/container/docker/templates/conductor-dockerfile.j2
+++ b/container/docker/templates/conductor-dockerfile.j2
@@ -1,9 +1,13 @@
 FROM {{ conductor_base }}
 ENV ANSIBLE_CONTAINER=1
 {% set distro = conductor_base.split(':')[0] %}
-{% if distro in ["fedora", "centos"] %}
+{% if distro in ["fedora"] %}
+RUN dnf update -y && \
+    dnf install -y gcc git python2 python2-devel rsync libffi-devel openssl-devel redhat-rpm-config python2-dnf && \
+    dnf clean all
+{% elif distro in ["centos"] %}
 RUN yum update -y && \
-    {% if distro in ["centos"] %}yum install -y epel-release && \{% endif %}
+    yum install -y epel-release && \
     yum install -y gcc git python-devel rsync libffi-devel openssl-devel && \
     yum clean all
 {% elif distro in ["debian", "ubuntu"] %}
